iml-system-test-utils/src/lib.rs

```diff
@@ -0,0 +1,68 @@
+pub mod docker;
+pub mod iml;
+pub mod vagrant;
+
+use futures::{future::BoxFuture, FutureExt, TryFutureExt};
+use std::{io, process::ExitStatus, time::Duration};
+use tokio::{process::Command, time::delay_for};
+
+fn handle_status(x: ExitStatus) -> Result<(), io::Error> {
+    if x.success() {
+        Ok(())
+    } else {
+        let err = io::Error::new(
+            io::ErrorKind::Other,
+            format!("process exited with code: {:?}", x.code()),
+        );
+        Err(err)
+    }
+}
+
+pub trait CheckedStatus {
+    fn checked_status(&mut self) -> BoxFuture<Result<(), io::Error>>;
+}
+
+impl CheckedStatus for Command {
+    /// Similar to `status`, but returns `Err` if the exit code is non-zero.
+    fn checked_status(&mut self) -> BoxFuture<Result<(), io::Error>> {
+        println!("Running cmd: {:?}", self);
+
+        self.status()
+            .and_then(|x| async move { handle_status(x) })
+            .boxed()
+    }
+}
+
+pub async fn try_command_n_times(max_tries: u32, cmd: &mut Command) -> Result<(), io::Error> {
+    let mut count = 0;
+    let mut r = cmd.status().await?;
```

Comment:
Because of this call, this fn actually tries the command n + 1 times. 

`count` should start at `1`, which will account for this.